src/errors.rs

```diff
@@ -55,7 +55,14 @@ quick_error! {
             description(err.description())
             display("protobuf error {:?}", err)
         }
-
+        /// The node exists, but should not.
+        Exists(id: u64, set: &'static str) {
```

Comment:
Can these two errors be merged?

Reply:
Hm, maybe? I think `Exists(u64, &'static str, bool)` could work? I worry that there might be some semantic issues. I was already not sure about combining things like `VoterCannotBecomeLearner`, `VoterExists`, `LearnerExists`, and `LearnerAlreadyVoter`.

Reply:
I think distinguishing them may be more clear.

Reply:
I don't think so. People usually don't care about what the exact case are they, what they do is just panic. More errors lead to longer match or repeated duplicated `error1 | error 2`.

Reply:
@BusyJay In the case of this particular error there are definitely recoverable cases, and they are different between the errors.

If the user is lazy and wants to panic, that's fine, but we should allow our users to write good code.

Reply:
I am fine with this, like Rust IO error, it has NotFound or AlreadyExists error for a file.

LGTM